### PR TITLE
feat: add agent-setup-validate command

### PR DESCRIPTION
**Pull request by torchello**

### CODE REVIEW

File: lib/appmap/service/validator/config_validator.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        def initialize(config_file)
+          @config_file = config_file
+          @violations = []
+          @config = load_config
+        end
+
+        def valid?
+          validate_config_presence
+          @violations.length == 0
```
Comment (kgilpin) [inline lib/appmap/service/validator/config_validator.rb:19]: Equivalently, `@violations.empty?`

File: lib/appmap/service/validator/config_validator.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        end
+
+        private
+
+        def present?
+          File.exist?(@config_file)
+        end
+
+        def load_config
+          AppMap::Config.load_from_file(@config_file) if present?
```
Comment (kgilpin) [inline lib/appmap/service/validator/config_validator.rb:29]: It's possible for other errors besides parse errors to come from this call. So, I'd suggest to split this into two phases.

First, `YAML.parse` the config and see if it's valid. If it's valid, `load(config_data)` and handle any exceptions (which would be unexpected, but might happen).

File: lib/appmap/service/validator/config_validator.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        rescue StandardError => e
+          @violations << Violation.error(
+            filename: @config_file,
+            message: 'AppMap configuration is not valid YAML',
```
Comment (kgilpin) [inline lib/appmap/service/validator/config_validator.rb:33]: The config file might be valid YAML here, but some other error occurred. That's why I suggest splitting this in two.

File: lib/appmap/service/validator/config_validator.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        attr_reader :config, :violations
+
+        def initialize(config_file)
+          @config_file = config_file
+          @violations = []
+          @config = load_config
+        end
+
+        def valid?
+          validate_config_presence
```
Comment (kgilpin) [inline lib/appmap/service/validator/config_validator.rb:18]: Are you going to check the Ruby version as well?

Comment (torchello) [reply lib/appmap/service/validator/config_validator.rb:19]: Planned to implement in a follow up PR, but it seems to be simple, so adding now.

File: lib/appmap/service/validator/config_validator.rb (hunk 2 above)
Comment (torchello) [reply lib/appmap/service/validator/config_validator.rb:31]: Makes sense, done!

**Review by kgilpin — APPROVED** [review]

**Pull request merged**